src/services/ui/src/features/package-actions/lib/modules/respond-to-rai/spa/med-rai.tsx

```diff
@@ -0,0 +1,53 @@
+import { z } from "zod";
+import {
+  zAdditionalInfo,
+  zAttachmentOptional,
+  zAttachmentRequired,
+} from "@/utils";
+import { ReactElement } from "react";
+import {
+  ActionDescription,
+  AdditionalInfoSection,
+  AttachmentsSection,
+  PackageSection,
+} from "@/components";
+
+export const medSpaRaiSchema = z.object({
+  additionalInformation: zAdditionalInfo.optional(),
+  attachments: z.object({
+    raiResponseLetter: zAttachmentRequired({ min: 1 }),
+    other: zAttachmentOptional,
+  }),
+});
+export const medSpaRaiFields: ReactElement[] = [
```

Comment:
So we are basically mapping over these lists to determine what we need to render for that form and reusing the common components as needed cause it is just JSX. Correct?

Reply:
Correct, and each component in this array is able to make use of `useFormContext` within it, so we have a "wireless" connection between ActionForm's zod schema plug-in and each individual field. This is the void between them where we structure them for render.